python/TestHarness/TestHarness.py

```diff
@@ -239,6 +242,8 @@ def findAndRunTests(self, find_only=False):
                 os.chdir(saved_cwd)
                 sys.path.pop()
     except KeyboardInterrupt:
+      if not self.options.failed_tests:
+        self.writeFailedTest.close()
```

Comment:
Not quite! You can ask the filehandle directly about it's state, you don't want to use an extra variable to do that because it can easily get out of sync.
```
if (!self.writeFailedTest.closed)
  self.writeFailedTest.close()
```